ipatests/test_integration/test_otp.py

```diff
@@ -392,6 +456,108 @@ def test_2fa_disable_single_prompt(self):
             self.master.run_command(['semanage', 'login', '-D'])
             sssd_conf_backup.restore()
 
+    def test_2fa_only_with_password(self):
+        """Test ssh with 2FA only with the password(first factor) when
+        user-auth-type is opt and password.
+
+        Test for :  https://github.com/SSSD/sssd/pull/7500
+
+        Add the IPA user and user-auth-type set to opt and password.
+        Authenticate the user only with password, just press enter
+        at `Second factor`
+        """
+
+        master = self.master
+        USER3 = 'sshuser3'
+        sssd_conf_backup = tasks.FileBackup(master, paths.SSSD_CONF)
+        first_prompt = 'Enter first factor:'
+        second_prompt = 'Enter second factor:'
+        add_contents = textwrap.dedent('''
+            [prompting/2fa/sshd]
+            single_prompt = False
+            first_prompt = {0}
+            second_prompt = {1}
+            ''').format(first_prompt, second_prompt)
+        set_sssd_conf(master, add_contents)
+        tasks.create_active_user(master, USER3, PASSWORD)
+        tasks.kinit_admin(master)
+        tasks.clear_sssd_cache(self.master)
+        master.run_command(['ipa', 'user-mod', USER3, '--user-auth-type=otp',
+                            '--user-auth-type=password'])
+        try:
+            otpuid, totp = add_otptoken(master, USER3, otptype='totp')
+            master.run_command(['ipa', 'otptoken-show', otpuid])
+            totp.generate(int(time.time())).decode('ascii')
+            otpvalue = "\n"
+            tasks.clear_sssd_cache(self.master)
+            sssd_version = tasks.get_sssd_version(master)
+            with xfail_context(sssd_version < tasks.parse_version('2.9.5'),
```

Comment:
The scenario is failing for SSSD = 2.9.5 on fedora but should succeed with SSSD = 2.9.5 on RHEL. The condition needs to be more specific:
with xfail_context((sssd_version < tasks.parse_version('2.9.5') and osinfo.id == 'rhel') or
                                 (sssd_version < tasks.parse_version('2.9.6') and osinfo.id == 'fedora')):